app/presenters/vote_button_presenter.rb

```diff
@@ -1,9 +1,71 @@
 class VoteButtonPresenter < ApplicationPresenter
-  def vote?
-    parents_vote.blank?
+  presents :parent
+
+  def vote
+    @vote ||= Vote.find_by(user_id: h.current_user, voteable_id: parent)
+  end
+
+  def buttons
+    if !h.user_signed_in?
+      static_icons
+    else
+      if vote.blank?
+        new_vote_sequence
+      elsif vote&.up_vote?
+        up_vote_sequence
+      elsif vote&.down_vote?
+        down_vote_sequence
+      end
+    end
+  end
+
+  def static_icons
+    render_partial("static_icons", nil)
+  end
+
+  def new_vote_sequence
+    active_up_button + active_down_button + delete_icon
+  end
+
+  def up_vote_sequence
+    up_icon + active_down_button + active_delete_button
+  end
+
+  def down_vote_sequence
+    active_up_button + down_icon + active_delete_button
+  end
+
+  def active_up_button
+    render_partial("up", { parent: parent })
+  end
+
+  def active_down_button
+    render_partial("down", { parent: parent })
+  end
+
+  def active_delete_button
+    render_partial("delete", { vote: vote })
+  end
```

Comment:
Is there a screenshot of the different states of these buttons? It would help me see what this logic is doing.

Reply:
sure @albaer - I just uploaded a gif to the PR comment, which illustrates the button states.